StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1502CodeFixProvider.cs

```diff
@@ -0,0 +1,221 @@
+
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1502ElementMustNotBeOnASingleLine"/>.
+    /// </summary>
+    [ExportCodeFixProvider(nameof(SA1502CodeFixProvider), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1502CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1502ElementMustNotBeOnASingleLine.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create("SA1502: Expand element", token => this.GetTransformedDocumentAsync(context, diagnostic, token)), diagnostic);
+            }
+
+            return Task.FromResult(true);
+        }
+
+        private async Task<Document> GetTransformedDocumentAsync(CodeFixContext context, Diagnostic diagnostic, CancellationToken token)
+        {
+            var syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var newDocument = this.CreateCodeFix(context, diagnostic, syntaxRoot);
+
+            return newDocument;
+        }
+
+        private Document CreateCodeFix(CodeFixContext context, Diagnostic diagnostic, SyntaxNode syntaxRoot)
+        {
+            SyntaxNode newSyntaxRoot = syntaxRoot;
+            var node = syntaxRoot.FindNode(diagnostic.Location.SourceSpan);
+            var indentationOptions = IndentationOptions.FromDocument(context.Document);
+
+            switch (node.Kind())
+            {
+                case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.InterfaceDeclaration:
+                case SyntaxKind.StructDeclaration:
+                    newSyntaxRoot = this.RegisterTypeDeclarationCodeFix(syntaxRoot, (TypeDeclarationSyntax)node, indentationOptions);
+                    break;
+
+                case SyntaxKind.AccessorList:
+                    newSyntaxRoot = this.RegisterPropertyLikeDeclarationCodeFix(syntaxRoot, (BasePropertyDeclarationSyntax)node.Parent, indentationOptions);
+                    break;
+
+                case SyntaxKind.Block:
+                    newSyntaxRoot = this.RegisterMethodLikeDeclarationCodeFix(syntaxRoot, (BaseMethodDeclarationSyntax)node.Parent, indentationOptions);
+                    break;
+
+                case SyntaxKind.EnumDeclaration:
+                    newSyntaxRoot = this.RegisterEnumDeclarationCodeFix(syntaxRoot, (EnumDeclarationSyntax)node, indentationOptions);
```

Comment:
By changing RegisterTypeDeclarationCodeFix to take a BaseTypeDeclarationSyntax you can remove the special handling of enum.